src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.File;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Deadline task that inherits from Task
+ */
+
+public class Deadline extends Task{
+
+    private String deadline;
+
+    public Deadline(String deadline, String task) {
+        super(task);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Returns the deadline specified by user.
+     *
+     * @return Deadline of Deadline task.
+     * @throws DateTimeParseException When user does not specify date of deadline in "MMM d yyyy" format.
+     */
+    public String getDeadline() throws DateTimeParseException {
+        try {
+            LocalDate ddl = LocalDate.parse(deadline);
+            return "(by: " + ddl.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        } catch (DateTimeParseException e) {
+            return "(by: " + deadline + ")";
+        }
+    }
+
+    /**
+     * Returns the String that specifies the Deadline task
+     * @return Deadline category.
+     */
+    public String isDeadline() {
```

Comment:
"isDeadline" signals that it might be returning a Boolean rather than a String. Perhaps a better name can be given...